src/views/AnnualReport.vue

```diff
@@ -506,33 +502,46 @@ export default {
       }
     }
 
-    // NB: filing id of 0 means "new AR"
-    // otherwise it's a draft AR filing id
-    this.filingId = +this.$route.params.filingId // number (may be NaN)
+    // Filing ID may be 0, N or NaN
+    this.filingId = +this.$route.params.filingId
+  },
 
+  mounted (): void {
     // if tombstone data isn't set, go back to dashboard
     if (!this.entityIncNo || !this.ARFilingYear || isNaN(this.filingId)) {
```

Comment:
I moved this check inside mounted so that we exit if needed (otherwise we were getting routing errors).

I also added more checks below for Coop ARs and BComp ARs.